languages/python/oso/tests/test_polar.py

```diff
@@ -948,3 +949,59 @@ def __init__(self, action):
 
     assert not is_allowed("leina", "write", r)
     assert is_allowed("leina", "read", r)
+
+
+def test_rule_prototypes_with_subclass_check(polar):
+    class Foo:
+        pass
+
+    class Bar(Foo):
+        pass
+
+    class Baz(Bar):
+        pass
+
+    class Bad:
+        pass
+
+    # NOTE: keep this order of registering classes--confirms that MROs are added at the correct time
+    polar.register_class(Baz)
+    polar.register_class(Bar)
+    polar.register_class(Foo)
+    polar.register_class(Bad)
+
+    p = """
+    type f(_x: Integer);
+    f(1);
+    """
+    polar.load_str(p)
+
+    p = """
+    type f(_x: Foo);
+    type f(_x: Foo, _y: Bar);
```

Comment:
For the future, it'd be cool to disable Singleton Var checks for rule prototypes so we can write this w/o the underscores

Reply:
Oh yeah that would definitely be good